docs/manual.md

```diff
@@ -807,91 +805,294 @@ user:
+* `"\\!foo"` - escape sequence, check if any line contains `!string`
+* `"/[Rr]egex/"` - verifies that line matches regex
+* `"!/[Rr]egex/"` - inverse of above, checks that no line matches regex
 
-**NOTE:** Regex support is based on golang's regex engine documented [here](https://golang.org/pkg/regexp/syntax/)
+**NOTE:** Regex support is based on Golang's regex engine documented [here](https://golang.org/pkg/regexp/syntax/)
 
 **NOTE:** You will **need** the double backslash (`\\`) escape for Regex special entities, for example `\\s` for blank spaces.
 
-### Example
-```bash
-$ cat /tmp/test.txt
-found
-!alsofound
+Example:
+
+```yaml
+file:
+  /tmp/test.txt:
+    exists: true
+    contains:
+    - "foo"
+    - "!bar"
+    - "/[Gg]oss/"
+```
 
+The above is roughly similar to:
 
-$ cat goss.yaml
+```yaml
 file:
   /tmp/test.txt:
     exists: true
     contains:
-    - found
-    - /fou.d/
-    - "\\!alsofound"
-    - "!missing"
-    - "!/mis.ing/"
+      and:
+        - contain-element: "foo"
+        - not: {contain-element: "bar"}
+        - contain-element: {match-regexp: "[Gg]oss"}
+
+```
+
+### Transforms
+
+If the system state type and the expected type don't match, goss will attempt to transform the system state type before matching it.
+
+For example, kernel-param attribute returns a string, however, it can be tested using numeric comparisons:
+
+Example kernel-param test:
+```yaml
+kernel-param:
+  net.core.somaxconn:
+      value: "128"
+```
+
+Kernel-param (failing) test with transform:
+```yaml
+kernel-param:
+  net.core.somaxconn:
+      value: {gt: 200}
+```
+
+When a transformed test fails, it will detail the transformers used, the `-o include_raw` option can be used to include the raw, untransformed attribute value:
+```
+$ goss v
+F
+
+Failures/Skipped:
+
+KernelParam: net.core.somaxconn: value:
+Expected
+    128
+to be >
+    200
+the transform chain was
+    [{"to-numeric":{}}]
+
+Total Duration: 0.001s
+Count: 1, Failed: 1, Skipped: 0
+
+
+$ goss v -o include_raw
+F
 
-$ goss validate
-..
+Failures/Skipped:
+
+KernelParam: net.core.somaxconn: value:
+Expected
+    128
+to be >
+    200
+the transform chain was
+    [{"to-numeric":{}}]
+the raw value was
+    "128"
 
 Total Duration: 0.001s
-Count: 2, Failed: 0
+Count: 1, Failed: 1, Skipped: 0
+
 ```
 
-## Advanced Matchers
-Goss supports advanced matchers by converting json input to [gomega](https://onsi.github.io/gomega/) matchers.
+### Advanced Matchers
 
-### Examples
+Goss supports advanced matchers by converting YAML input to [gomega](https://onsi.github.io/gomega/) matchers.
+
+#### String Matchers
 
-Validate that user `nobody` has a `uid` that is less than `500` and that they are **only** a member of the `nobody` group.
+These will convert the system attribute to a string prior to matching.
 
+* `'55'` - Checks that the numeric is "55" when converted to string
+* `have-prefix: pre` - Checks if string starts with "pre"
```

Comment:
This works because the default type in YAML is a string, right?

Reply:
you mean `have-prefix: pre` vs `have-prefix: "pre"` if so, yes.. that's yaml-ism.